app/recordtransfer/jobs.py

```diff
@@ -61,18 +61,23 @@ def create_downloadable_bag(submission: Submission, user_triggered: User) -> Non
                 new_job.save()
                 return
 
-        zipf = None
+        temp_zip_path = None
         try:
-            LOGGER.info("Zipping directory to an in-memory file ...")
-            zipf = BytesIO()
-            zipped_bag = zipfile.ZipFile(zipf, "w", zipfile.ZIP_DEFLATED, False)
-            zip_directory(submission.location, zipped_bag)
-            zipped_bag.close()
+            os.makedirs(settings.TEMP_STORAGE_FOLDER, exist_ok=True)
+            temp_fd, temp_zip_path = tempfile.mkstemp(
+                suffix=".zip", dir=settings.TEMP_STORAGE_FOLDER
+            )
+            os.close(temp_fd)
```

Comment:
You're closing the file and then re-opening it later, you may as well just keep it open and reuse that file handle later:

```suggestion
            temp_zip_fd, temp_zip_path = tempfile.mkstemp(
                suffix=".zip", dir=settings.TEMP_STORAGE_FOLDER
            )
```

Reply:
Addressed by: https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/445/commits/c22094c163fd4a821d29877ca087d0c0385bc4da